CNV/olm-install.sh

```diff
@@ -0,0 +1,244 @@
+#                     operator group globally 
+NAMESPACED_SUBSCR=${NAMESPACED_SUBSCR:-false}
+# WAIT_FOR_OBJECT_CREATION - Time in seconds for script to wait for some of the required objects to be created in Kubernetes. Time out will cause script to exit
+WAIT_FOR_OBJECT_CREATION=${WAIT_FOR_OBJECT_CREATION:-60}
+
+
+
+GLOBAL_NAMESPACE="${GLOBAL_NAMESPACE:-openshift-marketplace}"
+APP_REGISTRY="${APP_REGISTRY:-rh-osbs-operators}"
+PACKAGE="${PACKAGE:-kubevirt-hyperconverged}"
+TARGET_NAMESPACE="${TARGET_NAMESPACE:-openshift-cnv}"
+OPERATOR_NAME="${OPERATOR_NAME:-hco-operatorhub}"
+CHANNEL_VERSION="${CHANNEL_VERSION:-2.1.0}"
+SUBSCRIPTION_APPROVAL="${SUBSCRIPTION_APPROVAL:-Manual}"
+export TARGET_NAMESPACE
+
+
+if [[ ${CUSTOM_APPREGISTRY} ]]; then
+
+  QUAY_TOKEN="${QUAY_TOKEN:-}"
+
+  if [ -z "${QUAY_TOKEN}" ]; then 
+
+    # If application registry authentication token hasn't been provided check for username/password
+    QUAY_USERNAME="${QUAY_USERNAME:-}"
+    QUAY_PASSWORD="${QUAY_PASSWORD:-}"
+
+    if [ -z "${QUAY_USERNAME}" ]; then
+        echo ">>> QUAY_USERNAME is not set "
+        exit 1
+    fi
+
+    if [ -z "${QUAY_PASSWORD}" ]; then
+        echo ">>> QUAY_PASSWORD is not set "
+        exit 1
+    fi
+
+  else
+
+    echo ">>> Token for Quay registry has been provided. Validating access ..."
+
+    # To validate that token is correct and credentials are valid we will extract username and password from token. This step may look somewhat redundant but
+    # unfortunately at this point Quay doesn't seem to provide alternative API to validate access 
+
+    QUAY_USERNAME=$(echo ${QUAY_TOKEN}|cut -d' '  -f2|base64 -d |cut -d : -f1)
+    QUAY_PASSWORD=$(echo ${QUAY_TOKEN}|cut -d' '  -f2|base64 -d |cut -d : -f2)
+
+  fi
+
+  QUAY_TOKEN=$(curl -sH "Content-Type: application/json" -XPOST https://quay.io/cnr/api/v1/users/login -d '
+  {
+      "user": {
+          "username": "'"${QUAY_USERNAME}"'",
+          "password": "'"${QUAY_PASSWORD}"'"
+      }
+  }' | jq -r '.token')
+
+    if [ "${QUAY_TOKEN}" == "null" ]; then
+        echo "TOKEN was 'null'.  Did you enter the correct quay Username & Password?"
+        exit 1
+    fi
+
+    echo ">>> Creating registry secret"
+    cat <<EOF | oc apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: "quay-registry-${APP_REGISTRY}"
+  namespace: "${GLOBAL_NAMESPACE}"
+type: Opaque
+stringData:
+      token: "${QUAY_TOKEN}"
+EOF
+
+fi
+
+# 1. Create namespace for new operator 
+echo ">>> Create target namespace ${TARGET_NAMESPACE}"
+if ! `oc get project ${TARGET_NAMESPACE} &>/dev/null`;then
+    oc create ns ${TARGET_NAMESPACE}
+fi
+
+# 2. Create OperatorGroup defining namespaces that OLM will be monitoring  
+echo ">>> Creating operatorgroup ${TARGET_NAMESPACE}-group"
+if ! `oc get operatorgroup ${TARGET_NAMESPACE}-group -n ${TARGET_NAMESPACE} &>/dev/null` ; then
+  if [[ ${NAMESPACED_SUBSCR} ]]; then
+
+    echo "Creating OperatorGroup"
```

Comment:
The group is aslo created in the _else_ clause, but not mentioned.
Would be great to see a message about it there

Reply:
Good find , script has been updated